### PR TITLE
Enable logging in CI build and fix build warning from logging call

### PR DESCRIPTION
**Pull request by aggarw13**
* Fix build warning from a logging call.
* Enable logging in the `compiler-warnings` CI check to catch build issues from logging-related code.

### CODE REVIEW

**Review by yanjos-dev — DISMISSED** [review]

File: source/ota.c [hunk 1]
```diff
@@ -1979,7 +1979,7 @@ static OtaJobParseErr_t handleCustomJob( const char * pJson,
             /* Job is malformed - return an error */
             err = OtaJobParseErrNonConformingJobDoc;
 
-            LogError( ( "Custom job document parsing failed: OtaJobParseErr_t=%s, jobIdLength=%d",
+            LogError( ( "Failed to parse Custom job document: OtaJobParseErr_t=%s, jobIdLength=%lu",
```
Comment (leegeth) [inline source/ota.c:1982]:
```suggestion
            LogError( ( "Failed to parse custom job document: OtaJobParseErr_t=%s, jobIdLength=%lu",
```

**Review by leegeth — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
       - name: Check warnings
         run: |
           cmake -S test -B build -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS_INIT='-DLOG_LEVEL=LOG_LEVEL_DEBUG' \
-          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror'
+          -DCMAKE_C_FLAGS='-Wall -Wextra -Werror -DLOG_LEVEL=LOG_LEVEL_DEBUG'
```
Comment (aggarg) [inline .github/workflows/ci.yml:23]: Is `-DCMAKE_C_FLAGS_INIT='-DLOG_LEVEL=LOG_LEVEL_DEBUG'` still needed? This seems repeated.

Comment (aggarw13) [reply .github/workflows/ci.yml:23]: Yes, that is not needed and is not the correct CMake variable to set for defining the macro. I have removed it.

Comment (aggarg) [reply .github/workflows/ci.yml:23]: Thanks.

**Review by aggarg — APPROVED** [review]

**Review by yanjos-dev — APPROVED** [review]

**Pull request merged**